src/scoring/hpf_scoring.py

```diff
@@ -193,96 +166,65 @@ def match_manifest(self, input_id_set):  # pragma: no cover
-        :return companion_recommendation: The filtered list of recommended companion packages
-        along with condifence score.
+    def recommend_new_user(self, input_id_set):
+        """Implement the 'add_user' addon of HPFREC model. 
+
+        For more information please follow: https://hpfrec.readthedocs.io/en/latest/source/hpfrec.html
         """
-        companion_recomendations = []
-        input_stack = list(input_stack)
-        result = np.delete(result, input_stack)
-        result = np.where(np.isfinite(result))[0]
-        if self.USE_FEEDBACK:
-            alpha_id = int(self.match_feedback_manifest(set(input_stack)))
-            alpha_set = set(np.where(self.alpha[alpha_id] >= feedback_threshold)[0])
-            result = np.delete(result, list(alpha_set))
-
-        def _sigmoid(x):
-            return 1 / (1 + np.exp(-x))
-        recommended_ids = result.argsort()[::-1][:MAX_COMPANION_REC_COUNT]
-        mean = np.mean(result[recommended_ids])
-        result = result - mean
-        result = np.divide(result, np.std(result))
-        for package_id in recommended_ids:
-            recommendation = {
-                "cooccurrence_probability": round(_sigmoid(result[package_id]) * 100, 2),
-                "package_name": self.id_package_dict[package_id],
-                "topic_list": []  # We don't have topics for this ecosystem!!
-            }
-            # At least thirty percent probability is required for recommendation to go through.
-            if recommendation['cooccurrence_probability'] > MIN_REC_CONFIDENCE:
-                companion_recomendations.append(recommendation)
-        return companion_recomendations
+        new_df = pd.DataFrame({
+                'ItemId':input_id_set,
+                'Count': [1]*len(input_id_set)})
+        user_id  =self.recommender.nusers
+        is_user_added = self.recommender.add_user(user_id = user_id, counts_df = new_df)
+        if is_user_added:
+            user_id -= 1
+            recommendations = self.recommender.topN(user_id, n = self.m)
+            return recommendations, user_id
+        else:
+            _logger.info('Unable to add user')
+
+
+    def filter_recommendation(self, result, input_stack, user_id):
+        """Used for calculaing cooccurrence probability of companion packages and for filter packages."""
+        package_id_set = input_stack
+        recommendations = result
+        companion_packages = []
+        recommendations = np.array(list(itertools.compress(recommendations,
+                                        [i not in package_id_set for i in recommendations])))
+
+        print("Filtered recommendation ids are: " + str(recommendations))
+
+        poisson_values = self.recommender.predict(
+            user=[user_id] * recommendations.size,
+            item=recommendations
+        )
+
+        def sigmoid(array):
+            return 1/(1 + np.exp(-array))
```

Comment:
Leave a couple of blank lines after the end of function. You can use Pycharm auto formatting to actually format your code (Use Ctrl + Alt + L).